src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -209,7 +212,7 @@ private static string ComputeHash(string data, string algorithm)
         {
             // Disable MD5 insecure warning.
 #pragma warning disable CA5351
-            using (HashAlgorithm hash = algorithm.Contains(Sha256) ? SHA256.Create() : (HashAlgorithm)MD5.Create())
+            using (HashAlgorithm hash = algorithm.StartsWith(Sha256, StringComparison.OrdinalIgnoreCase) ? SHA256.Create() : (HashAlgorithm)MD5.Create())
```

Comment:
And here the logic is different too. Deliberate?

Reply:
Yes, same reason as above, only 4 values are currently allowed. This is correct check, but for the 4 allowed values `Contains` and `StartsWith` have the same result (except for the case insensitivity).